viewer/packages/tools/src/SmartOverlay/SmartOverlayTool.ts

```diff
@@ -80,121 +77,59 @@ export class SmartOverlayTool extends Cognite3DViewerToolBase {
     this._textOverlay = this.createTextOverlay('', this._defaultTextOverlayToCursorOffset);
     viewer.domElement.appendChild(this._textOverlay);
 
-    this._singeOverlayCollection = new Overlay3DCollection(undefined, {
-      maxPointSize: toolParameters?.maxPointSize
-    });
-    this._overlayPoints.push(this._singeOverlayCollection);
-    viewer.addObject3D(this._singeOverlayCollection);
-
     viewer.canvas.addEventListener('mousemove', this.onMouseMove);
 
     viewer.on('click', this.onMouseClick);
   }
 
   /**
-   * Adds multiple overlay
-   * @param overlays Array of labels to add.
+   * Creates new OverlayCollection.
+   * @param overlays Array of overlays to add.
    * @returns Overlay group containing it's id.
    */
-  addOverlays(overlays: OverlayInfo[]): OverlayGroup {
+  createOverlayCollection(overlays: OverlayInfo<MetadataType>[]): OverlayCollection<MetadataType> {
```

Comment:
Should `overlays` be an optional argument here, as it is to `Overlay3DCollection`?

Reply:
Yeah, that's right